### ISSUE TITLE
When groupData is set to true and selectperiod  is set to default as 1month its loading correct in android but showing wrong data in ios  

### ISSUE THREAD

**Issue by VanavarayanK** [amcharts/amcharts5#719]
android app
![image](https://user-images.githubusercontent.com/118241594/207674953-023d64c8-dcba-4747-a9e4-ecc777324929.png)

in ios app it loads data
![image](https://user-images.githubusercontent.com/118241594/207773040-6470ccdf-9981-4538-9773-14e458f20350.png)

**Comment by martynasma** [comment]
Would you be able to post your whole chart and data on CodePen/jsFiddle so that we can test?

**Comment by VanavarayanK** [comment]
Hi 
https://jsfiddle.net/5fbn8eam/10/
please refer to this link 

Also sometimes the last date is not showing at the end

**Comment by VanavarayanK** [comment]
Hi Team,

Thanks for reaching out and please refer to this link
https://jsfiddle.net/5fbn8eam/10/

Also sometimes the last date is not showing at the end

Regards,
Vanavarayan

**Comment by martynasma** [comment]
The jsFiddle you posted works good for me now. Or is this related to https://github.com/amcharts/amcharts5/issues/719?

**Comment by VanavarayanK** [comment]
Yes, In web browsers it works fine whereas if we try to use this  in mobile
application or ios browers from mobile u will get to know the difference as
above image,only  IOS users facing this issue. Also in JSfiiddle forcefully
i have added a active class  by using below code,  that should not be the
case if we give selectPeriod as 1month then it should take automatically
that is not happening right now.

let divID = document.getElementsByClassName("am5stock-link")
 divID[1].className += "   am5stock-active";

**Comment by martynasma** [comment]
OK, so I'm trying to understand what is wrong with the images that you posted.

This is what I'm getting on desktop:
<img width="305" alt="chrome_2022-12-19_12-19-57" src="https://user-images.githubusercontent.com/1458520/208404047-bfed9edb-cbee-48ee-941a-6c1aef507daf.png">

And this is in Safari, in an iPhone simulator:
![Simulator Screen Shot - iPhone 12 Pro Max - 2022-12-19 at 12 20 28](https://user-images.githubusercontent.com/1458520/208404170-5dbcf6f6-817a-4955-b700-7f5b647446fd.png)

Both display Dec 14 as the last item, with the same sum.

Please explain what exactly is wrong in the iOS.

Please also check whether you are running latest version.

**Comment by VanavarayanK** [comment]
Yes i am running the latest version and  last date is not an issue for me,
start date is the issue when its one month showing 15th Nov to 14 Dec in
android but in ios that is not happening some dates getting cropped in the
beginning.

**Comment by martynasma** [comment]
OK, so it looks some kind of timing issue in iOS.

If I delay `selectPeriod()` call by a fraction of second, the chart comes up fine:

```JavaScript
    valueSeries.events.on("datavalidated", function(err,data) {
      setTimeout(function() {
        periodSelector.selectPeriod({ timeUnit: "month", count: 1, name: "1M" })
      }, 100);
    })
```

I'll investigate if this is something we can fix on out end.

In the interim, feel free to use the delay workaround.

**Comment by VanavarayanK** [comment]
Thanks for the workaround suggested and your time. It works for some of the data but it is failing for some and I'm facing the same issue, It would be of great help if you could look into this.

**Comment by martynasma** [comment]
Fixed in [5.2.43](https://github.com/amcharts/amcharts5/releases/tag/5.2.43).

## [5.2.43] - 2022-12-19

### Added
- An event `groupintervalchanged` added to `DateAxis`. Kicks in when data grouping is on, and current group interval changes, e.g. via zooming the chart.

### Fixed
- Setting `min` and `max` on a `DateAxis` which was not base axis of a series could result in a stack overflow error in some specific cases.
- `PeriodSelector` was not always highlighting the period button when period was selected via `selectPeriod()` method.
- `PeriodSelector` was not selecting proper start position on some platforms when data grouping was on and `selectPeriod()` was called immediately after initialization of the chart.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by VanavarayanK** [comment]
Thanks a lot.
